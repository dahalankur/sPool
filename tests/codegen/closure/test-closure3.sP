# nested functions
def quack hi():
    int i = -99
    println("Inside hi()")
    def quack bye(int i):
        print("Inside bye with arg i = ")
        println(int_to_string(i)) # shadowing captured i with the argument
        return;
    print("Inside hi with variable i = ")
    println(int_to_string(i))
    bye(55)
    return
;

hi()


# TODO: lambdas, higher order functions (passing functions as args, returning functions from functions, assigning functions to variables, etc.)