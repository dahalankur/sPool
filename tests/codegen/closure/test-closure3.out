Inside hi()
Inside hi with variable i = -99
Inside bye with arg i = 55
